src/bus.c

```diff
@@ -116,6 +93,52 @@ uint8_t bus_spi_read_one(uint8_t cs, uint8_t reg) {
     return buf[0];
 }
 
+void bus_i2c_init() {
+    printf("INIT: I2C bus\n");
+    i2c_init(i2c1, I2C_FREQ);
+    gpio_set_function(PIN_SDA, GPIO_FUNC_I2C);
+    gpio_set_function(PIN_SCL, GPIO_FUNC_I2C);
+    gpio_pull_up(PIN_SDA);
+    gpio_pull_up(PIN_SCL);
+    if (!gpio_get(PIN_SDA) || !gpio_get(PIN_SCL)) {
+        printf("ERROR: I2C bus is not clean\n");
+        sleep_ms(1000000000);
+    }
+}
+
+void bus_i2c_io_init_single(uint8_t id) {
+    bus_i2c_write(id, I2C_IO_REG_POLARITY,   0b11111111);
+    bus_i2c_write(id, I2C_IO_REG_POLARITY+1, 0b11111111);
+    bus_i2c_write(id, I2C_IO_REG_PULL,   0b11111111);
+    bus_i2c_write(id, I2C_IO_REG_PULL+1, 0b11111111);
+    printf("  IO id=%i ", id);
+    printf("ack=%i ", bus_i2c_acknowledge(I2C_IO_0));
+    printf("polarity=%i ", bin(bus_i2c_read_one(I2C_IO_0, I2C_IO_REG_POLARITY)));
+    printf("pull=%i\n", bin(bus_i2c_read_one(I2C_IO_0, I2C_IO_REG_PULL)));
```

Comment:
These statements use hard-coded I2C device address `I2C_IO_0`. I expect it should use provided function argument instead.

```suggestion
    printf("ack=%i ", bus_i2c_acknowledge(id));
    printf("polarity=%i ", bin(bus_i2c_read_one(id, I2C_IO_REG_POLARITY)));
    printf("pull=%i\n", bin(bus_i2c_read_one(id, I2C_IO_REG_PULL)));
```

Reply:
Nice catch